cli/internal/core/engine.go

```diff
@@ -161,10 +159,29 @@ func (e *Engine) generateTaskGraph(pkgs []string, taskNames []string, tasksOnly
 
 		topoDeps := util.SetFromStrings(task.TaskDefinition.TopologicalDependencies)
 
+		deps := make(util.Set)
+		isPackageTask := util.IsPackageTask(taskName)
+
+		for _, dependency := range task.TaskDefinition.TaskDependencies {
+			// If the current task is a workspace-specific task (including root Task)
+			// and its dependency is _also_ a workspace-specific task, we need to add
+			// a reference to this dependency directly into the engine.
+			// TODO @mehulkar: Why do we need this?
+			if isPackageTask && util.IsPackageTask(dependency) {
+				if err := e.AddDep(dependency, taskName); err != nil {
+					return err
+				}
```

Comment:
This part is kind of strange, hopefully we can make some improvements so that packageTasks don't have a separate code path some time in the future. I'm not sure they need to be tracked separately. (Especially now that we're doing this work per-workspace.